File: src/LiteGraph/TransactionResult.cs
namespace LiteGraph
{
    using System;
    using System.Collections.Generic;

    /// <summary>
    /// Graph-scoped transaction result.
    /// </summary>
    public class TransactionResult
    {
        #region Public-Members

        /// <summary>
        /// True if the transaction committed.
        /// </summary>
        public bool Success { get; set; } = true;

        /// <summary>
        /// True if the transaction rolled back.
        /// </summary>
        public bool RolledBack { get; set; } = false;

        /// <summary>
        /// Index of failed operation, if any.
        /// </summary>
        public int? FailedOperationIndex { get; set; } = null;

        /// <summary>
        /// Error message, if the transaction failed.
        /// </summary>
        public string Error { get; set; } = null;

        /// <summary>
        /// Operation results.
        /// </summary>
        public List<TransactionOperationResult> Operations { get; set; } = new List<TransactionOperationResult>();

        /// <summary>
        /// Transaction duration in milliseconds.
        /// </summary>
        public double DurationMs { get; set; } = 0;

        #endregion
    }
}

File: sdk/js/src/models/TransactionResult.js
export class TransactionOperationResult {
  constructor(result = {}) {
    const {
      Index = 0,
      OperationType = 'Create',
      ObjectType = 'Node',
      GUID = null,
      Success = true,
      Result = null,
      Error = null,
    } = result;

    this.Index = Index;
    this.OperationType = OperationType;
    this.ObjectType = ObjectType;
    this.GUID = GUID;
    this.Success = Success;
    this.Result = Result;
    this.Error = Error;
  }
}

export default class TransactionResult {
  constructor(result = {}) {
    const {
      Success = true,
      RolledBack = false,
      FailedOperationIndex = null,
      Error = null,
      Operations = [],
      DurationMs = 0,
    } = result;

    this.Success = Success;
    this.RolledBack = RolledBack;
    this.FailedOperationIndex = FailedOperationIndex;
    this.Error = Error;
    this.Operations = Operations.map((operation) => new TransactionOperationResult(operation));
    this.DurationMs = DurationMs;
  }
}
